external-crates/move/crates/move-vm-runtime/src/shared/vm_pointer.rs

```diff
@@ -139,3 +131,29 @@ impl<T> From<Box<T>> for VMPointer<T> {
         VMPointer(raw_ptr)
     }
 }
+
+impl<T: PartialOrd> PartialOrd for VMPointer<T> {
```

Comment:
Nit: might be nice to add a brief comment on why these are not pointer based (e.g., non-determinism for `Ord`, etc)